web/src/components/vis/PcaPage/PcaPage.vue

```diff
@@ -36,25 +39,45 @@ export default {
   },
 
   computed: {
-    pcX() {
-      return Number.parseInt(this.pcXval, 10);
-    },
-
-    pcY() {
-      return Number.parseInt(this.pcYval, 10);
-    },
-
-    numComponents() {
-      return Number.parseInt(this.numComponentsVal, 10);
-    },
-
     ready() {
       const pcaReady = this.$store.getters.ready(this.id, 'pca');
       const loadingsReady = this.$store.getters.ready(this.id, 'loadings');
 
       return pcaReady && loadingsReady;
     },
   },
+  watch: {
+    pcXval: {
+      handler(val) {
+        const pcX = Number.parseInt(val, 10);
+        // eslint-disable-next-line no-restricted-globals
+        if (!isNaN(pcX)) {
+          this.pcX = pcX;
+        }
+      },
+      immediate: true,
+    },
+    pcYval: {
+      handler(val) {
+        const pcY = Number.parseInt(val, 10);
+        // eslint-disable-next-line no-restricted-globals
+        if (!isNaN(pcY)) {
```

Comment:
`Number.isNaN()`  should fix the lint error?

Reply:
https://stackoverflow.com/questions/46677774/eslint-unexpected-use-of-isnan

The behavior varies slightly from `isNaN`, but since the inputs are only `number` or `NaN` guaranteed by parseInt, you should be safe.

Reply:
What about `window.isNaN()`?